h/static/scripts/virtual-thread-list.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+var EventEmitter = require('tiny-emitter');
+var debounce = require('lodash.debounce');
+var inherits = require('inherits');
+
+/**
+ * VirtualThreadList is a helper for virtualizing the annotation thread list.
+ *
+ * 'Virtualizing' the thread list improves UI performance by only creating
+ * annotation cards for annotations which are either in or near the viewport.
+ *
+ * Reducing the number of annotation cards that are actually created optimizes
+ * the initial population of the list, since annotation cards are big components
+ * that are expensive to create and consume a lot of memory. For Angular
+ * applications this also helps significantly with UI responsiveness by limiting
+ * the number of watchers (functions created by template expressions or
+ * '$scope.$watch' calls) that have to be run on every '$scope.$digest()' cycle.
+ *
+ * @param {Window} container - The Window displaying the list of annotation threads.
+ * @param {Thread} rootThread - The initial Thread object for the top-level
+ *        threads.
+ */
+function VirtualThreadList($scope, window_, rootThread) {
```

Comment:
`VirtualThreadList` is instantiated directly (both in `widget-controller.js` and in the tests) rather than via Angular's injector so we don't need to use the `$window` service here.